pkg/gadgets/run/tracer/tracer.go

```diff
@@ -870,6 +930,12 @@ func (t *Tracer) AttachContainer(container *containercollection.Container) error
 		}
 	}
 
+	for _, handler := range t.tcHandlers {
+		if err := handler.AttachContainer(container); err != nil {
+			return err
+		}
```

Comment:
If an interfaces was specified with "iface" param, we must not call this (or do nothing inside the tchandler) to avoid incorrectly running the gadget also for the containers. If I'm didn't missing anything or misunderstood the iface flag behaviour, this is a bug.

Reply:
Right, it was the last TODO item of this PR. I changed the logic not to attach to containers if this parameter is specified.